spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+  describe "GET #index" do
+    subject(:index) { get :index }
+
+    it { is_expected.to be_successful }
+  end
+
+  describe "GET #show" do
+    subject(:show) { get :show, params: { id: article.id } }
+    let(:article) { create(:article) }
+
+    it { is_expected.to be_successful }
+  end
+
+  describe "POST #create" do
+  context "when the article is valid" do
```

Comment:
złe wcięcia w obu testach, Railsy się o to nie kłócą, ale inne frameworki mogą, więc warto z góry wyrabiać dobre nawyki

Reply:
To zawsze robił za mnie program w innych frame xd